stim/config.go

```diff
@@ -19,119 +23,226 @@ func (stim *Stim) Get(configKey string) interface{} {
+		return nil, err
 	}
+	return config, err
+}
 
-	return nil
+func (stim *Stim) ConfigIsCustom() bool {
+	cfp, _ := filepath.Abs(stim.ConfigGetString("config-file"))
+	return defaultStimConfigFilePath != cfp
 }
 
-func (stim *Stim) GetStimPath() (string, error) {
-	home, err := homedir.Dir()
+func (stim *Stim) ConfigGetStimConfigDir() (string, error) {
+	cfp, err := stim.ConfigGetStimConfigFile()
 	if err != nil {
 		return "", err
 	}
-	return home + "/.stim/", nil
+	dir, _ := path.Split(cfp)
+	return dir, nil
 }
 
 // CreateConfigFile will create the stim config file if it doesn't exist
 // Used the frist time this code is ran so sub functions do not get errors when
 // writting to the config.
-func (stim *Stim) CreateConfigFile() (string, error) {
-	home, err := stim.GetStimPath()
+func (stim *Stim) ConfigGetStimConfigFile() (string, error) {
+	cfp, err := filepath.Abs(stim.ConfigGetString("config-file"))
+	custom := defaultStimConfigFilePath != cfp
 	if err != nil {
 		return "", err
 	}
-	stimConfigFile := home + "config.yaml"
-
-	dir, _ := path.Split(stimConfigFile)
-	err = stim.CreateDirIfNotExist(dir)
-	if err != nil {
+	_, err = os.Stat(cfp)
+	if err != nil && !os.IsNotExist(err) {
 		return "", err
+	} else if err != nil && custom && os.IsNotExist(err) {
+		return "", err
+	} else if err == nil {
+		return cfp, nil
 	}
-	newFile, err := os.Create(stimConfigFile)
+	err = utils.CreateFileIfNotExist(cfp, utils.UserOnlyMode)
 	if err != nil {
 		return "", err
 	}
-	newFile.Close()
-
-	return stimConfigFile, nil
-}
-
-func (stim *Stim) CreateDirIfNotExist(dir string) error {
-	stim.log.Debug("Creating config path: ", dir)
-	if _, err := os.Stat(dir); os.IsNotExist(err) {
-		err := os.MkdirAll(dir, 0755)
-		if err != nil {
-			return err
-		}
-	}
-
-	return nil
+	return cfp, nil
 }
 
-func (stim *Stim) loadConfigFile() error {
-
+func (stim *Stim) configLoadConfigFile() error {
 	// Set the config file type
 	stim.config.SetConfigType("yaml")
-
 	// Don't forget to read config either from CfgFile or from home directory!
-	configFile := stim.GetConfig("config-file")
-	_, err := os.Stat(configFile)
-	if err != nil && !os.IsExist(err) && !stim.IsAutomated() {
-		stim.log.Warn("No config file exists at :\"" + configFile + "\"")
-		//If they passed in a custom path we might want to exit here
+	configFile, err := stim.ConfigGetStimConfigFile()
+	if err != nil {
+		stim.log.Warn("{}", err)
+		if stim.ConfigIsCustom() {
+			stim.log.Fatal("Problem loading config file from custom path:'{}', exiting!", configFile)
+		} else {
+			stim.log.Warn("Problem loading config file:'{}', continuing using ENV", configFile)
+			//We return no error here since its not always a problem to not have a config file
+			return nil
+		}
 	}
 	stim.config.SetConfigFile(configFile)
 	confErr := stim.config.ReadInConfig()
 	return confErr
 }
+
+func (stim *Stim) configInitDefaultValues() {
+	if stim.ConfigIsCustom() {
```

Comment:
Why don't we want to set default values if the config path is custom?

Reply:
I feel like it would be used more for custom scripts and what not.  I am gonna remove the writing of defaults in either case for now anyway so it wont matter much for now.